frame/contracts/primitives/src/lib.rs

```diff
@@ -105,6 +105,23 @@ pub enum ContractAccessError {
 	DoesntExist,
 	/// Storage key cannot be decoded from the provided input data.
 	KeyDecodingFailed,
+	/// Storage is migrating. Try again later.
+	MigrationInProgress,
+}
+
+impl<T, B: Zero, EventRecord> From<DispatchError>
+	for ContractResult<Result<T, DispatchError>, B, EventRecord>
+{
+	fn from(error: DispatchError) -> Self {
+		ContractResult {
+			gas_consumed: Zero::zero(),
+			gas_required: Zero::zero(),
+			storage_deposit: Default::default(),
+			debug_message: Vec::new(),
+			result: Err(error),
+			events: None,
+		}
+	}
 }
```

Comment:
this is what used for all the 
```rust
		if let Err(err) = Migration::<T>::ensure_migrated() {
			return err.into()
		}
```
Would you prefer a Migration::<T>::is_migrated() with a  ContractResult::migration_in_progress()

Reply:
Yeah I think we should not have this `From` impl. Even though it makes this code uglier. I think it would make it to easy to return this default struct for other errors where we don't wan this.